zappa/core.py

```diff
@@ -2920,6 +2920,12 @@ def get_scheduled_event_name(self, event, function, lambda_name, index=0):
         Returns an AWS-valid CloudWatch rule name using a digest of the event name, lambda name, and function.
         This allows support for rule names that may be longer than the 64 char limit.
         """
+        function_regex = re.compile("^[._A-Za-z0-9]+$")
```

Comment:
Can you add a link in the docstring to the rule name requirement?
https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-resource-events-rule.html

![image](https://user-images.githubusercontent.com/1135529/205469958-f0318e07-5791-43ed-a8c3-363098c959fc.png)

The expression is shown as:

> [\.\-_A-Za-z0-9]+

Or is this just for the function name?

You can also add min/max with {1, 64} here instead of using "+".

> ^[\.\-_A-Za-z0-9]{1,64}$

https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html
![image](https://user-images.githubusercontent.com/1135529/205470071-357323fa-f783-445e-9b0a-513b59073611.png)

Reply:
> Can you add a link in the docstring to the rule name requirement?
OK!